$NetBSD: patch-eh,v 1.1 2011/09/19 07:15:52 dholland Exp $

- silence warning about assignment in an if condition
- fix parenthesization, caught by clang

--- clients/xmtg/pathfind.c~	1997-05-21 17:15:13.000000000 +0000
+++ clients/xmtg/pathfind.c
@@ -72,7 +72,7 @@ pathfind
       if (i) {
          if ((dirp = opendir(pbuf)) == 0)
             continue;
-         while (dep = readdir(dirp)) {
+         while ((dep = readdir(dirp)) != NULL) {
             if (strcmp(name, dep->d_name) == 0) {
                pbuf[i] = '\0';		/* reset path */
                (void)strcat(pbuf, "/");
@@ -86,25 +86,25 @@ pathfind
                for (mp=mode; *mp && okay; mp++) {
                   switch (*mp) {
                      case 'r':
-                        if (	st.st_mode & S_IROTH == 0 &&
-				(st.st_mode & S_IRUSR == 0 || uid != st.st_uid))
-                           if (st.st_mode & S_IRGRP == 0 || gid != st.st_gid)
+                        if (	(st.st_mode & S_IROTH) == 0 &&
+				((st.st_mode & S_IRUSR) == 0 || uid != st.st_uid))
+                           if ((st.st_mode & S_IRGRP) == 0 || gid != st.st_gid)
                               okay = 0;
                            else
                               chkgrp = 1;
                         break;
                      case 'w':
-                        if (	st.st_mode & S_IWOTH == 0 &&
-				(st.st_mode & S_IWUSR == 0 || uid != st.st_uid))
-                           if (st.st_mode & S_IWGRP == 0 || gid != st.st_gid)
+                        if (	(st.st_mode & S_IWOTH) == 0 &&
+				((st.st_mode & S_IWUSR) == 0 || uid != st.st_uid))
+                           if ((st.st_mode & S_IWGRP) == 0 || gid != st.st_gid)
                               okay = 0;
                            else
                               chkgrp = 1;
                         break;
                      case 'x':
-                        if (	st.st_mode & S_IXOTH == 0 &&
-				(st.st_mode & S_IXUSR == 0 || uid != st.st_uid))
-                           if (st.st_mode & S_IXGRP == 0 || gid != st.st_gid)
+                        if (	(st.st_mode & S_IXOTH == 0) &&
+				((st.st_mode & S_IXUSR == 0) || uid != st.st_uid))
+                           if ((st.st_mode & S_IXGRP == 0) || gid != st.st_gid)
                               okay = 0;
                            else
                               chkgrp = 1;
@@ -125,13 +125,13 @@ pathfind
                         okay = S_ISFIFO(st.st_mode);
                         break;
                      case 'u':
-                        okay = (st.st_mode & S_ISUID != 0);
+                        okay = ((st.st_mode & S_ISUID) != 0);
                         break;
                      case 'g':
-                        okay = (st.st_mode & S_ISGID != 0);
+                        okay = ((st.st_mode & S_ISGID) != 0);
                         break;
                      case 'k':
-                        okay = (st.st_mode & S_ISVTX != 0);
+                        okay = ((st.st_mode & S_ISVTX) != 0);
                         break;
                      case 's':
                         okay = (st.st_size != 0);
